low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -214,6 +215,56 @@ def get_platform_sensors_info(self):
                 )
         return response
 
+    def get_sas_hba_info(self):
+        """Return SAS-HBA current health."""
+        sas_hba_data = []
+        hosts = self.sysfs.get_sas_host_list()  # ['host1']
+        for host in hosts:
+            host_id = SAS_RESOURCE_ID + host.replace('host', '')
+            host_data = self.get_health_template(host_id, False)
+            ports = self.sysfs.get_sas_port_list(host)
+            # ports = ['port-1:0', 'port-1:1', 'port-1:2', 'port-1:3']
+            health = "OK"
+            specifics = {
+                'num_ports': len(ports),
+                'ports': []
+            }
+            for port in ports:
+                port_data = self.sysfs.get_sas_port_data(port)
+                specifics['ports'].append(port_data)
+                if port_data['state'] != 'running':
+                    health = "NA"
+
+            self.set_health_data(host_data, health, specifics=[specifics])
+            sas_hba_data.append(host_data)
+        return sas_hba_data
+
+    def get_sas_ports_info(self):
+        """Return SAS Ports current health."""
+        sas_ports_data = []
+        ports = self.sysfs.get_sas_port_list()
+        # eg: ['port-1:0', 'port-1:1', 'port-1:2', 'port-1:3']
+        for port in ports:
+            port_id = 'sas_' + port
```

Comment:
Ensure this follows health schema. 
I don't see "last_updated" attribute added to this data.

Reply:
ensured 👍